src/pages/iou/request/step/withWritableReportOrNotFound.tsx

```diff
@@ -0,0 +1,55 @@
+import type {RouteProp} from '@react-navigation/core';
+import type {ComponentType, ForwardedRef, RefAttributes} from 'react';
+import React, {forwardRef} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import getComponentDisplayName from '@libs/getComponentDisplayName';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
+import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Report} from '@src/types/onyx';
+
+type WithWritableReportOrNotFoundOnyxProps = {
+    /** The report corresponding to the reportID in the route params */
+    report: OnyxEntry<Report>;
+};
+
+type Route = RouteProp<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT>;
+
+type WithWritableReportOrNotFoundProps = WithWritableReportOrNotFoundOnyxProps & {route: Route};
+
+export default function <TProps extends WithWritableReportOrNotFoundProps, TRef>(
+    WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
+): React.ComponentType<Omit<TProps & RefAttributes<TRef>, keyof WithWritableReportOrNotFoundOnyxProps>> {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithWritableReportOrNotFound(props: TProps, ref: ForwardedRef<TRef>) {
+        const {report, route} = props;
```

Comment:
`report` has default value (empty json) in original logic.
Leaving it as `undefined` causes regression:

<img width="325" alt="Screenshot 2024-03-21 at 4 25 42 PM" src="https://github.com/Expensify/App/assets/108292595/30593215-9ae4-4d50-b741-708139082bbb">

Reply:
@VickyStash @fabioh8010 So this is why the default value was needed 😂 

I'll add it right away @situchan! Thanks for finding this 😄